src/contrib/cluster/Akka.DistributedData/Serialization/SerializationSupport.cs

```diff
@@ -101,19 +105,26 @@ public static byte[] Decompress(byte[] input)
                 using(var inputStr = new MemoryStream(input))
                 using (var gzipStream = new GZipStream(inputStr, CompressionMode.Decompress))
                 {
-                    var buf = new byte[BufferSize];
-                    while (gzipStream.CanRead)
+                    var buf = BufferPool.Rent(BufferSize);
```

Comment:
Byte array allocation optimization. Use a memory pool instead of allocating new byte array all the time.